subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -98,9 +99,11 @@ static int settings_set(const char *key, size_t len_rd,
 			settings_read_cb read_cb, void *cb_arg)
 {
 	if (!strcmp(key, FILE_FLASH_IMG)) {
-		ssize_t len = read_cb(cb_arg, &flash_img.bytes_written,
-				      sizeof(flash_img.bytes_written));
-		if (len != sizeof(flash_img.bytes_written)) {
+		size_t bytes_written = flash_img_bytes_written();
```

Comment:
```suggestion
		size_t bytes_written = flash_img_bytes_written(&flash_img);
```

Reply:
Anything using dfu_target_mcuboot.c does not build without changing this.

Reply:
Thanks @sigvartmh. Fixed now. I'm not sure what you mean by your last sentence though.